compaction/task_manager_module.cc

```diff
@@ -664,3 +655,22 @@ future<> shard_resharding_compaction_task_impl::run() {
 }
 
 }
+
+auto fmt::formatter<compaction::flush_mode>::format(compaction::flush_mode fm, fmt::format_context& ctx) const -> decltype(ctx.out()) {
+    std::string_view name;
+    switch (fm) {
+    using enum compaction::flush_mode;
+    case skip:
+        name = "skip";
+        break;
+    case compacted_tables:
+        name = "compacted_tables";
+        break;
+    case all_tables:
+        name = "all_tables";
+        break;
+    default:
```

Comment:
If you don't define the default case, we would get an error in compile time rather than runtime if another enum value is added.

Reply:
i wanted to preserve the existing behavior. i am not 100% sure that the compiler is able to populate the constant here and make it a compile-timer failure. if that's the case, it would be great. but i don't really understand your point here. do you mean we should handle the `default`case?

Reply:
no, we shouldn't handle it since the compiler should complain if we there would a new class enum value that's not handled by the switch statement (try it)

Reply:
ahh, i see. you are talking about the `-Wswitch` warning. and you are suggesting we should use it as an indication that we might actually fail to handle some case(s). so this is more future-proof. but with `default:`, compiler would consider all cases are covered, so this warning won't show up.

ack. removed.